src/observables.jl

```diff
@@ -37,3 +37,53 @@ rydberg_corr(reg) = rydberg_corr(Op.n, reg)
 function rydberg_corr(op, reg)
     return [expect(chain(nqubits(reg), put(i => op), put(j => op)), reg) for i in 1:nqubits(reg), j in 1:nqubits(reg)]
 end
+
+
+"""
+    get_average_rydberg_densities(atoms, reg; [C=2π * 862690 * MHz*µm^6], Ω[, ϕ, Δ], [dt=1e-3 * μs])
+
+Return average Rydberg densities throughout an evolution
+
+# Arguments
+
+- `atoms`: a collection of atom positions.
+- `reg`: required, the register object.
+
+# Keyword Arguments
+
+- `C`: optional, default unit is `MHz*µm^6`, interation parameter,
+    see also [`RydInteract`](@ref).
+- `Ω`: optional, default unit is `MHz`, Rabi frequencies, divided by 2, see also [`SumOfX`](@ref).
+- `ϕ`: optional, does not have unit, the phase, see [`SumOfXPhase`](@ref).
+- `Δ`: optional, default unit is `MHz`, detuning parameter, see [`SumOfN`](@ref).
+- `dt`: optional, default unit is `μs`, time step for the evolution
+"""
+function get_average_rydberg_densities(atoms, reg; C = 2π * 862690, Ω = nothing, ϕ = nothing, Δ = nothing, dt=1e-3)
+
+    # Get the duration for the evolution
+    if isnothing(Ω) && isnothing(ϕ) && isnothing(Δ)
+        error("At least one of Ω, ϕ or Δ is needed for determining the duration of the evolution.")
+    end
+
+    allwaveforms = [Ω, ϕ, Δ]
+    allwaveforms = allwaveforms[allwaveforms.!=nothing]
+    allwaveforms = reduce(vcat, allwaveforms)
+    duration = allwaveforms[1].duration
+    for i = 2 : length(allwaveforms)
+        if allwaveforms[i].duration != duration
+            error("The durations of waveforms are not consistent.")
+        end
+    end
+
+    # Start the evolution 
+    h = rydberg_h(atoms, C=C, Δ=Δ, Ω=Ω, ϕ=ϕ)
+    prob = SchrodingerProblem(reg, duration, h, progress=true);
+    integrator = init(prob, Vern8());
```

Comment:
Can we make the Vern8 solver as a keyword?

Reply:
Done, but could you specify the type of the `solver`? Thanks!

Reply:
I think we will leave the solver type here, since there are too many different kinds of solvers.

Reply:
Ok, then I will keep this file as it is now. I will add a few tests in the next couple of days.